modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/util/DropdownItem.java

```diff
@@ -18,6 +20,10 @@ public void setIcon(String icon) {
 		put("icon", icon);
 	}
 
+	public void setItems(JSONArray itemsJSONArray) {
+		put("items", itemsJSONArray);
+	}
+
```

Comment:
We already have a [DropdownGroupItem](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/util/DropdownGroupItem.java#L13) that handles the nesting. Is there any reason not to reuse it for this use case?

Reply:
Thanks Sandro, I want to make sure we don't introduce a duplicate abstraction here. The fact that we pass type=group makes the grouping case different from the nested one. I'd say approach is ok

Reply:
@sandrodw3 @markocikos don't we like to hoist up `setDropdownItems()` method from [DropdownGroupItem](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/util/DropdownGroupItem.java#L19-L21) up to `DropdownItem` instead of adding a new method on the parent class?

Note this method takes a `List<DropdownItem> dropdownItems` argument, more suitable to add items as result of `DropdownItemListBuilder`. Moreover, child class `DropdownGroupItem` is inheriting 2 different methods to add items, which looks odd.

Reply:
Illustrated [here](https://github.com/dsanz/liferay-portal/commits/LPD-25954-with-example/), from code courtesy of @sandrodw3 who kindly sent https://github.com/dsanz/liferay-portal/pull/84